### PR TITLE
Proposed fix for release build failure due to not including junit4-stub dependency

### PR DESCRIPTION
**Pull request by phantomjinx**

### CODE REVIEW

**Comment by phantomjinx** [comment]
retest this please

File: app/pom.xml [hunk 1]
```diff
@@ -322,6 +322,7 @@
         <basepom.check.skip-all>true</basepom.check.skip-all>
         <basepom.it.skip>true</basepom.it.skip>
         <skipTests>true</skipTests>
+        <maven.test.skip>true</maven.test.skip>
```
Comment (zregvart) [inline app/pom.xml]: Not sure we need this, compiler, resource, surefire and failsafe are removed from running below, so this doesn't make a difference.

File: app/pom.xml [hunk 2]
```diff
@@ -371,22 +372,6 @@
...
-            <executions>
-              <execution>
-                <id>attach-sources</id>
-                <goals>
-                  <goal>jar</goal>
-                </goals>
-              </execution>
-            </executions>
-          </plugin>
-
```
Comment (zregvart) [inline app/pom.xml]: Don't we want to release sources?

File: app/pom.xml [hunk 3]
```diff
@@ -603,17 +588,6 @@
...
-            <groupId>com.mycila</groupId>
-            <artifactId>license-maven-plugin</artifactId>
-            <executions>
-              <execution>
-                <id>basepom.default</id>
-                <phase />
-              </execution>
-            </executions>
-          </plugin>
-
```
Comment (zregvart) [inline app/pom.xml]: This disables license header check, why run that if we're not running any other checks?

File: app/s2i/pom.xml [hunk 4]
```diff
@@ -187,7 +187,36 @@
...
+                </licenseUrlReplacement>
+              </licenseUrlReplacements>
+              <skipAggregateDownloadLicenses>false</skipAggregateDownloadLicenses>
+              <executeOnlyOnRootModule>false</executeOnlyOnRootModule>
+              <licensesOutputDirectory>${image.build.directory}/licenses</licensesOutputDirectory>
+              <licensesOutputFile>${image.build.directory}/licenses.xml</licensesOutputFile>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
```
Comment (zregvart) [inline app/s2i/pom.xml]: This restores the `license-maven-plugin` which will try to resolve test dependencies, and thus junit4-stub, which hasn't been built (not part of the reactor).

**Review by zregvart — CHANGES_REQUESTED** [review]
Yeah, I don't think this will help, most likely you had the `junit4-stub` installed in your local Maven repository.

File: app/s2i/pom.xml (hunk 4 above)
Comment (phantomjinx) [reply app/s2i/pom.xml]: Back to drawing board.

File: app/pom.xml (hunk 3 above)
Comment (phantomjinx) [reply app/pom.xml]: Restoring....

File: app/pom.xml (hunk 2 above)
Comment (phantomjinx) [reply app/pom.xml]: Obviously removed the wrong plugin. These is a duplicate error  as there is 2 maven-source-plugin in the release profile.

File: app/pom.xml (hunk 1 above)
Comment (phantomjinx) [reply app/pom.xml]: Thought this might solve the problem but sadly not.

**Comment by phantomjinx** [comment]
retest this please

**Pull request merged**